examples/nextjs/pages/e2ee.tsx

```diff
@@ -34,22 +36,67 @@ const E2EEExample: NextPage = () => {
               }
             : undefined,
       }),
-    [],
+    [keyProvider],
   );
 
-  room.setE2EEEnabled(true);
+  const tokenSource = useMemo(() => {
+    return TokenSource.endpoint(process.env.NEXT_PUBLIC_LK_TOKEN_ENDPOINT!);
+  }, []);
+
+  const session = useSession(tokenSource, {
+    roomName,
+    participantIdentity: userIdentity,
+    participantName: userIdentity,
+    room,
```

Comment:
question: @1egoman is there anything ensuring that session is still "in charge" of the room connection cycle when users use this pattern?

Reply:
Not currently, no. It's an "escape hatch" you can use to provide a custom room and if the caller modifies the room's state extensively out of band it will definitely cause things to break. This definitely needs to be documented better than it currently is.